components/brave_education/resources/brave_education.ts

```diff
@@ -0,0 +1,63 @@
+/* Copyright (c) 2024 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+import { BrowserCommandProxy } from 'chrome://resources/js/browser_command/browser_command_proxy.js';
+import { EducationPageProxy } from './education_page_proxy';
+import { readCommandMessage } from './message_data_reader'
+
+function pageReady() {
+  return new Promise<void>((resolve) => {
+    if (document.readyState === 'loading') {
+      document.addEventListener('DOMContentLoaded', () => resolve())
+    } else {
+      resolve()
+    }
+  })
+}
+
+async function handleMessage(data: any) {
+  // Read the message provided by the website. If the message is invalid, then
+  // an error will be thrown and will appear on the console for debugging.
+  const message = readCommandMessage(data)
+
+  // Check to make sure that the command can be executed for the current
+  // profile.
+  const { handler } = BrowserCommandProxy.getInstance()
+  const { canExecute } = await handler.canExecuteCommand(message.command)
```

Comment:
can we make it so that each link can ONLY do the commands specified for it in https://docs.google.com/document/d/1UKKGddj_-pwzLUpsXyRpVByCowHBvZRHLx0C5LxkAj4/edit ?

for instance https://brave.com/welcome/adblock/ can open content filter settings but not privacy settings.

Reply:
Yes - I think this is doable.